Updates to Fasta generation

axelwalter opened the pull request:
This PR should fix some details regarding Fasta file generation.

- remove variants with empty sequence
- join variant details & consequences with , in headers
- discard variants with inner stop gained
- check for existing uniprot accession for header
- extend frameshift flanking regions until the end

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

axelwalter committed:
do not splice frameshift variants to the left

axelwalter committed:
flanking regions in frameshift extent until the end

axelwalter committed:
remove debug prints

axelwalter committed:
trim whitspace

jonasscheid commented on bin/epaa.py:
```diff
@@ -297,7 +297,7 @@ def read_vcf(filename, pass_only=True):
                                 split_coding_c[0] if split_coding_c[0] else split_annotation[vep_fields["feature"]]
                             )
-                            transcript_id = transcript_id.split(".")[0]
+                            transcript_id = transcript_id #.split(".")[0]
```
transcript_id = transcript_id ?

axelwalter committed:
log exceptions

axelwalter committed:
increase robustness
if a variant annotation string is missing, skip variant instead of aborting
if no valid cds position, skip variant
sorting variant annotations in multi-variant peptides, check len

axelwalter committed:
remove transcript id version number

axelwalter committed:
add test for fasta output

axelwalter committed:
update sample sheet file path in test

axelwalter committed:
update CHANGELOG

axelwalter committed:
update fasta output flag description with suggested VEP parameters

axelwalter committed:
update fasta test config

axelwalter committed:
add fasta test to nextflow.config

jonasscheid committed:
Merge branch 'dev' into fasta-fix

axelwalter committed:
add fasta_output.nf.test file

axelwalter committed:
fix syntax in nextflow.config

jonasscheid committed:
Add missing profile specification in nf-test scripts, update snapshots
